lake-framework/src/types.rs

```diff
@@ -156,3 +171,13 @@ pub enum LakeError {
     #[error("Internal error: {error_message}")]
     InternalError { error_message: String },
 }
+
+// pub trait LakeMiddleware: Middleware +  {}
+
+#[async_trait::async_trait]
+pub trait LakeMiddleware: Debug + Send + Sync {
+    async fn process(
+        &self,
+        block: near_lake_primitives::block::Block,
+    ) -> near_lake_primitives::block::Block;
```

Comment:
Why do we need to return the `block`? can we just take it by reference instead?

Reply:
Sorry, I should have added a comment. It's temporary, WIP. There will be something like `impl Block` to allow "modifying the structure"